drivers/gpio/wch_gpio_ch32v00x.c

```diff
@@ -111,13 +113,137 @@ static int gpio_ch32v00x_port_toggle_bits(const struct device *dev, uint32_t pin
 	return 0;
 }
 
+#if defined(CONFIG_GPIO_WCH_GPIO_INTERRUPTS)
+
+static void gpio_ch32v00x_isr(uint8_t line, void *user)
+{
+	const struct device *dev = user;
+	struct gpio_ch32v00x_data *data = dev->data;
+
+	gpio_fire_callbacks(&data->callbacks, dev, BIT(line));
+}
+
+static int gpio_ch32v00x_configure_exti(const struct device *dev, gpio_pin_t pin)
+{
+	const struct gpio_ch32v00x_config *config = dev->config;
+	AFIO_TypeDef *afio = (AFIO_TypeDef *)DT_REG_ADDR(DT_NODELABEL(pinctrl));
+	uint8_t port_id;
+	uint8_t cr_id;
+	uint8_t bit0;
+
+	/* Convert the device into a port ID by checking the address */
+	switch ((uintptr_t)config->regs) {
+	case DT_REG_ADDR(DT_NODELABEL(gpioa)):
+		port_id = 0;
+		break;
+#if DT_NODE_EXISTS(DT_NODELABEL(gpiob))
+	case DT_REG_ADDR(DT_NODELABEL(gpiob)):
+		port_id = 1;
+		break;
+#endif
+	case DT_REG_ADDR(DT_NODELABEL(gpioc)):
+		port_id = 2;
+		break;
+	case DT_REG_ADDR(DT_NODELABEL(gpiod)):
+		port_id = 3;
+		break;
+	default:
```

Comment:
Some Ch32V30x have gpioe as well, I would recommend adding it as well, it is supported in the pinctrl driver with `#if DT_NODE_EXISTS(DT_NODELABEL(gpioe))`

Reply:
Done